stock_picking_product_assortment/views/stock_picking_view.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_picking_form" model="ir.ui.view">
+        <field name="name">stock.picking.form</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <field name="partner_id" position="after">
+                <field name="assortment_product_ids" invisible="1" />
+                <field name="has_assortment" invisible="1" />
+            </field>
+            <xpath
+                expr="//field[@name='move_ids_without_package']/list/field[@name='product_id']"
+                position="attributes"
+            >
+                <attribute
+                    name="domain"
+                    operation="domain_add"
+                    condition="parent.has_assortment"
+                >
+                    [('id', 'in', parent.assortment_product_ids)]
```

Comment:
@bosd This should not be solved here, but have you tested this with resulting large domains? This will work fine and well for a small set, but for large results... an `('id', 'in', parent.assortment_product_ids)` is really inefficient.
Is there no way to invert the domain like `('assortment_id', 'in', parent.assortment_ids)`?

Reply:
@PieterPaulussen I did not test this module with large domains or data sets.
I've tried the code suggestion, but it did'nt work yet.